### ISSUE TITLE
Kiali serving blank pages with Istio 1.0.6

### ISSUE THREAD

**Issue by mabushey** [kiali/kiali#845]
**Describe the bug**
Web GUI is a blank page (pre, body, and html tags, no rendered content). 
```
✗ kubectl -n istio-system logs -f kiali-6b5d46d49c-dj2qs
I0215 00:38:56.572583       1 kiali.go:63] Kiali: Version: v0.14.0, Commit: a1afa57fa0b0760a5a76108e94a31e53dd40c6c9
I0215 00:38:56.572849       1 kiali.go:84] Kiali: Console version: 0.14.0
I0215 00:38:56.572863       1 update_base_url.go:13] Updating base URL in index.html with [/kiali]
I0215 00:38:56.573785       1 javascript_config.go:13] Generating env.js from config
I0215 00:38:56.576129       1 server.go:44] Server endpoint will start at [:20001/kiali]
I0215 00:38:56.576142       1 server.go:45] Server endpoint will serve static content from [/opt/kiali/console]
I0215 00:38:56.576214       1 server.go:50] Server endpoint will require https
2019/02/15 01:06:45 http: TLS handshake error from 100.96.3.172:38720: tls: first record does not look like a TLS handshake
2019/02/15 01:06:45 http: TLS handshake error from 100.96.3.172:38722: tls: first record does not look like a TLS handshake
2019/02/15 01:06:52 http: TLS handshake error from 100.96.3.172:38776: tls: first record does not look like a TLS handshake
```

I don't want TLS. Maybe if down the road if Istio ever gets MutualTLS to work, but it's broke; we can't use it. SSL termination at the ELB makes me happy as it works, and works well.
 
**Versions used**
Kiali: 2019-02-14 getLatestKialiKubernetes.sh
Istio: 1.0.6
Kubernetes flavour and version:
Rocky Road?? Installed with Kops.
1.11.7

**To Reproduce**
export GRAFANA_URL="https://grafana.green.example.com"
export JAEGER_URL="https://trace.green.example.com"
./getLatestKialiKubernetes.sh - enter username and password
Apply:
```
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: kiali
  namespace: default
spec:
  hosts:
  - "kiali.example.com"
  - "kiali.blue.example.com"
  - "kiali.green.example.com"
  gateways:
  - elb-gateway
  http:
  - match:
    route:
    - destination:
        port:
          number: 32749
        host: kiali.istio-system.svc.cluster.local
```

**Expected behavior**
For it to be available at https://kiali.green.example.com

What's the reason for the NodePort? It should be ClusterIP like the rest of the Istio services.

**Comment by mabushey** [comment]
Ugg. <sarcasm>What an engaging community</sarcasm>

**Comment by cfcosta** [comment]
@mabushey hello, and thanks for reporting this bug! we must have missed this one, and sorry about that, things get lost sometimes because of the many priorities.

I'll reopen this and ask someone to take a look at it. It might take until monday because of the weekend.

**Comment by jmazzitelli** [comment]
This is reporting on 0.14.0. We are on 0.20.0. I believe the NodePort issue is now addressed since 0.19.0 - you can make service type ClusterIP or whatever you want by specifying it in the Kiali CR (used by the new operator):

https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml#L110-L113

You can also specify additional yaml to apply in the service to support additional data needed for certain service types. See: https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml#L74-L81

Install the operator, but do not install Kiali (because you do not want the defaults). Create a Kiali CR with the appropriate config you want and install the Kiali CR which will tell the operator to install Kiali with those settings.

See https://www.kiali.io/gettingstarted for details on installing the operator and kiali.

**Comment by lucasponce** [comment]
@mabushey Sorry for the noise with the bots.
Please, let us know if the last comments helped to address the issue.

Also, don't hesitate to ping us on #kiali channel on IRC freenode https://webchat.freenode.net/ if for some reason the issue was not updated.
